src/ModelSingleton.py

```diff
@@ -172,17 +173,22 @@ def __call__(self, query_text):
 
 
 class SapbertModelWrapper(ModelWrapper):
-    def __init__(self, model_path, all_reps_path, all_reps_ids_path):
+    """
+        host: "http://localhost:9200"
+    username: "elastic"
+    password: ""
+    index: "sap_index"
+    """
+    def __init__(self, model_path, elastic_search_config):
         """ Initializes NLP Model"""
         super(SapbertModelWrapper, self).__init__()
         self.tokenizer = AutoTokenizer.from_pretrained(model_path)
         self.model = AutoModel.from_pretrained(model_path).cuda(0)
-        self.all_reps_emb = np.load(all_reps_path)
-        all_reps_name_ids = pd.read_csv(all_reps_ids_path, header=0, dtype=str)
-        self.all_reps_names = all_reps_name_ids['Name']
-        self.all_reps_ids = all_reps_name_ids['ID']
+        self.elastic_client = SAPElastic(
+            **elastic_search_config
+        )
 
-    def __call__(self, query_text, count):
+    async def __call__(self, query_text, count=10, similarity="cosine", bl_type=""):
```

Comment:
Should we use knn as the default similarity measure method since knn is faster than cosine?